projects/solace-message-client/src/lib/ɵsolace-message-client.ts

```diff
@@ -580,7 +644,6 @@ export class ɵSolaceMessageClient implements SolaceMessageClient, OnDestroy {
             }
           }),
           take(1),
-          takeUntil(this._sessionDisposed$),
```

Comment:
Why we dont need this any more?

Reply:
Because we would not receive the 'DISCONNECTED' event otherwise.

When disposing the session, we emit `sessionDisposed$` immediately, canceling any listener waiting in `whenEvent`. But, the 'DISCONNECTED' event is fired later by the broker. For example, calling 'disconnect' to disconnect gracefully from the broker would never resolve and put the message client in an inconsistent state. I found out about that problem in our tests.